lib/tcpip/network_wrapper.cc

```diff
@@ -574,6 +645,34 @@ SObj network_socket_create_and_bind(Timeout       *timeout,
 				    mallocCapability, socket_key(), sealedSocket);
 				  return nullptr;
 			  }
+
+			  if (isListening)
+			  {
+				  auto listenResult =
+				    FreeRTOS_listen(socket, MAX_SIMULTANEOUS_TCP_CONNECTIONS);
```

Comment:
Sounds good, seems that there could be legitimate use-cases for this.

Reply:
More thoughts about "Maybe put that in the bind-and-listen capability?":

I view the purpose of bind and connection capabilities as a way to enforce least privilege at the network stack API.

Thus, I need to understand which privilege we would be enforcing by putting the "maximum number of TCP connections that can be created (by remote peers) into a given server port" into the bind capability.

Is the idea to limit how many TCP connections a compartment can create in total? I can indeed see a problem where compartment A prevents compartment B from establishing network connections, by using up all the TCP connection slots in the network stack (which is quite limited in practice, typically <10). But that problem already exists pretty badly in the network stack, since connection capabilities do not come with such a limit: they can be used as many times as a compartment wants to open as many connections as they want.

This makes me think that this is a broader issue in the API, and that we may want to have a fix directly dedicated at this problem that fixes both bind and connection capabilities together. It may not be trivial to fix to match real-world use-cases... I don't know if the limit should be global for one compartment, or local to each connection / bind capability. The latter may be difficult to use in practice.

Reply:
As we discussed online, let's put that maximum value in the bind capability and solve the problem separately for the connection capability: https://github.com/CHERIoT-Platform/network-stack/issues/45